pkg/controllers/node/node_syncer.go

```diff
@@ -37,6 +40,23 @@ func (c *NodeController) initSyncer() {
 
 func (c *NodeController) OnStatusUpdated(status bapi.SyncStatus) {
 	logrus.Infof("Node controller syncer status updated: %s", status)
+	c.syncStatus = status
+
+	switch status {
+	case bapi.ResyncInProgress:
+		// Starting the resync so clear our node cache.
+		c.nodeCache = make(map[string]*api.Node)
+
+	case bapi.InSync:
+		err := c.syncAutoHostendpoints()
+		exitOnError(err)
+	}
+}
+
+func exitOnError(err error) {
+	if err != nil {
+		os.Exit(1)
```

Comment:
Let's do `log.WithError(err).Fatal` for this for bonus context when exiting.

Reply:
269de076c49fcf1f59c14c96c0c242afcd5b7510